contracts/merkles/MerkleDrop.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+pragma solidity 0.7.5;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/IMerkleDrop.sol";
+
+
+/**
+ * @title MerkleDrop
+ *
+ * @dev MerkleDrop contract allows users to claim their tokens by proving that they're part of the merkle tree.
+ * Adopted from https://github.com/Uniswap/merkle-distributor/blob/0d478d722da2e5d95b7292fd8cbdb363d98e9a93/contracts/MerkleDistributor.sol
+ */
+contract MerkleDrop is IMerkleDrop, Ownable {
+    using SafeERC20 for IERC20;
+
+    // @dev Address of the token contract.
+    IERC20 public immutable override token;
+
+    // @dev Merkle Root for proving tokens ownership.
+    bytes32 public immutable override merkleRoot;
+
+    // @dev Expire timestamp for te merkle drop.
+    uint256 public immutable override expireTimestamp;
+
+    // This is a packed array of booleans.
+    mapping(uint256 => uint256) public override claimedBitMap;
+
+    /**
+    * @dev Constructor for initializing the MerkleDrop contract.
+    * @param _owner - address of the contract owner.
+    * @param _token - address of the token contract.
+    * @param _merkleRoot - address of the merkle root.
+    * @param _duration - duration of the merkle drop in seconds.
+    */
+    constructor(address _owner, address _token, bytes32 _merkleRoot, uint256 _duration) {
+        token = IERC20(_token);
+        merkleRoot = _merkleRoot;
+        // solhint-disable-next-line not-rely-on-time
+        expireTimestamp = block.timestamp + _duration;
+        transferOwnership(_owner);
+    }
+
+    /**
+     * @dev See {IMerkleDrop-isClaimed}.
+     */
+    function isClaimed(uint256 index) public view override returns (bool) {
```

Comment:
why are public and view required here? Do all view functions require the public keyword, isn't that redundant? (Still learning the language)

Reply:
The `public` here means that the function can be executed both by the other functions within this contract as well as by everyone outside this contract. If you check the `claim` function it calls the `isClaimed`. If that wouldn't be the case, we could make that function `external`. A good explanation can also be found here: https://ethereum.stackexchange.com/a/19391/66599